Improve alternative origin docs (#6095)
dfinity/portal: docs/building-apps/authentication/alternative-origins.mdx

## Configuring alternative origins

For this example, you will have two domains, **A** and **B**. **A** will be the derivation origin, and **B** will be the alternative domain. To help illustrate this model, consider this website, which is hosted both at `https://xxxxx.icp0.io` and `https://www.yourcustomdomain.com`.

In this example, **A** would be `https://xxxxx.icp0.io`.

**B** would be the alternative origin, or `https://www.yourcustomdomain.com`.

### Setting the derivation origin

> On the code for the **B** origin, you need to set the `derivationOrigin` to the AuthClient to **A**.

Author: I'll try to rephrase it, I introduced the setup before:

> consider this website, which is hosted both at https://xxxxx.icp0.io and https://www.yourcustomdomain.com.
> In this example, A would be https://xxxxx.icp0.io.
> 
> B would be the alternative origin, or https://www.yourcustomdomain.com.

The developer might want users to authenticate from both or only one, that's up to them...

```typescript
const authClient = await AuthClient.login({
  //...
  derivationOrigin: "https://xxxxx.icp0.io",
});
```

For more details about the [login options in the AuthClient, see its documentation](https://js.icp.build/auth/latest/api/client/interfaces/authclientloginoptions#derivationorigin).

### Listing alternative origins

On origin **A**, you need to provide an endpoint `.well-known/ii-alternative-origins` that tells Internet Identity that **B** is a valid alternative origin.

The response of this endpoint must be a valid JSON object with a `alternativeOrigins` key, containing an array of strings:

```json
{
  "alternativeOrigins": ["https://www.yourcustomdomain.com"]
}
```

#### Using the asset canister

> If you are using the assets canister. You'll be placing the config files in the `src/assets` directory of your frontend canister.

Reviewer: ```suggestion
If you are using the assets canister, you should place the config files in the `src/assets` directory of your frontend canister.
```

> If you are using the assets canister. You'll be placing the config files in the `src/assets` directory of your frontend canister.

Author: I changed the structure, let me know what you think. I added a whole section for "Asset Canister" with an introduction.
